tutor4points/core/views.py

```diff
@@ -94,16 +94,49 @@ def users(request, id):
 
     # get user that is specified by URL
     user = get_user_model().objects.get(pk=id)
-    if request.method == 'POST':
+    if request.method == 'POST' and 'email' in request.POST:
         form_update_profile = UpdateProfileForm(request.POST, request.FILES, instance=user)
         if form_update_profile.is_valid():
             form_update_profile.save()
     else:
         form_update_profile = UpdateProfileForm(instance=user)
 
+    if request.method == 'POST' and 'rating' in request.POST:#code for rating, move once 'paid and done' functionality is added.
+        form_rating = RateTutorForm(request.POST)
+        if form_rating.is_valid():
+            rating = form_rating.save(commit=False)
+            rating.given_to = user
+            rating.given_by = request.user
+            rating.save()
+
+            # calcute new rating, and update the database
+            type = form_rating.cleaned_data['rating_type']
+
+            if type == 'tutor':#for tutors
+                tutor_ratings = Rating.objects.filter(given_to = user).filter(rating_type = 'tutor')
+                count = 0
```

Comment:
Python has a mean() function. You might want to check it out. I'm not sure if that would help make the code more concise here.